### PR TITLE
Charmhub bases

### PR DESCRIPTION
**Pull request by SimonRichardson**
Although the API still accepts a platform for refresh API, it's actually
not the right way to do it. Instead, we should push bases all the way
through the stack. That way we know that this is actually future-proofed
against any API changes.

The following code just does that.

## QA steps

```sh
$ juju bootstrap lxd test
$ juju deploy hello-juju
$ juju deploy facundo-snappass-test
$ juju deploy ubuntu-juju-qa --channel 2.0/edge
```

### CODE REVIEW

File: apiserver/facades/client/charms/charmhubrepo.go [hunk 1]
```diff
@@ -276,7 +274,11 @@ func (c *chRepo) selectNextBase(bases []transport.Base, origin corecharm.Origin)
 		return corecharm.Platform{}, errors.NotFoundf("base")
 	}
 
-	series, err := coreseries.VersionSeries(channelTrack(base.Channel))
+	track, err := channelTrack(base.Channel)
+	if err != nil {
+		return corecharm.Platform{}, errors.Trace(err)
+	}
+	series, err := coreseries.VersionSeries(track)
```
Comment (manadart) [inline apiserver/facades/client/charms/charmhubrepo.go:281]: I still think we need to check how CentOS and Windows charms will appear as bases. Seems to me we are unlikely to get `win10` or `centos7` as a track. Something like `name: "windows"` and `track: "10"` would be the correct representation.

Comment (SimonRichardson) [reply apiserver/facades/client/charms/charmhubrepo.go:281]: There are currently no way to tell, as we don't have any pre-existing charms to test.

**Review by manadart — APPROVED** [review]
Working as expected for this iteration.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**